src/main/scala/io/iohk/ethereum/jsonrpc/BlockResponse.scala

```diff
@@ -61,6 +68,17 @@ object BlockResponse {
 
     val (lcn, td) = weight.map(_.asTuple).separate
 
+    val signature =
+      if (block.header.extraData.length >= ECDSASignature.EncodedLength)
```

Comment:
Just a minor comment, when we have the condition (block.header.extraData.length =  ECDSASignature.EncodedLength) 
The following line:
block.header.extraData.takeRight(ECDSASignature.EncodedLength) is not required, right?
Anyway, this code is clear and understandable.